module/userExtension/helper.js

```diff
@@ -211,304 +212,373 @@ module.exports = class UserExtensionHelper {
-                  existingUser.roles.push(userRoleMap[userRole.role]);
-                }
+                  if (!entry) {
+                    entry = { programId: programObjectId, roles: [] };
+                    userInformation.platformRoles.push(entry);
+                  }
 
-                existingUser.roles[userRoleToUpdate].entities = existingUser.roles[userRoleToUpdate].entities.map(
-                  (eachEntity) => eachEntity.toString(),
-                );
-
-                if (userRole.entityOperation == 'OVERRIDE') {
-                  existingUser.roles[userRoleToUpdate].entities = [userRole.entity];
-                  existingUser.roles[userRoleToUpdate].acl = userRole.acl;
-                } else if (userRole.entityOperation == 'APPEND' || userRole.entityOperation == 'ADD') {
-                  existingUser.roles[userRoleToUpdate].entities.push(userRole.entity);
-                  existingUser.roles[userRoleToUpdate].entities = _.uniq(existingUser.roles[userRoleToUpdate].entities);
-                } else if (userRole.entityOperation == 'REMOVE') {
-                  _.pull(existingUser.roles[userRoleToUpdate].entities, userRole.entity);
-                  removeUserFromEntity = true;
+                  for (const role of roles) {
+                    if (!entry.roles.includes(role)) {
+                      entry.roles.push(role);
+                      // Emit create event for new role
+                      kafkaEventPayloads.push({
+                        userId: userProfile.id,
+                        username: userProfile.username,
+                        programId: programObjectId,
+                        role,
+                        eventType: 'create',
+                      });
+                    }
+                  }
                 }
-
-                existingUser.roles[userRoleToUpdate].entities = existingUser.roles[userRoleToUpdate].entities.map(
-                  (eachEntity) => ObjectId(eachEntity),
-                );
-                updateQuery['roles'] = existingUser.roles;
               }
 
-              if (userRole.programOperation) {
-                let userPlatformRoleToUpdate;
+              user = await database.models.userExtension.create(userInformation);
+              userExtensionMap[user.userId.toString()] = user;
+
+              userRole['_SYSTEM_ID'] = user?._id || '';
+              userRole.status = user ? 'Success' : 'Failed to create the user role.';
+              userRole._kafkaEventPayloads = kafkaEventPayloads;
+            } else {
+              // Update existing user
+              let existingUserPlatformRoles = [...(existingUser.platformRoles || [])];
 
-                if (existingUser.platformRoles && existingUser.platformRoles.length > 0) {
-                  userPlatformRoleToUpdate = _.findIndex(existingUser.platformRoles, { code: userRole.platform_role });
-                }
+              if (userRole.programOperation && userRole.programs) {
```

Comment:
@borkarsaish65 appreciate some description comments for loops

Reply:
done